quinn/src/connection.rs

```diff
@@ -641,6 +641,17 @@ impl Connection {
         // May need to send MAX_STREAMS to make progress
         conn.wake();
     }
+
+    /// Track changed on our external address as reported by the peer.
+    // NOTE: this is currently leaking an implementation detail, but quinn does not use common deps
+    // like futures, so no access to the Stream future for example. Other utilities are also not
+    // imported, so for the sake of not incurring in bloat, but prevent the impl detail to leak, we
+    // should at least check on alternative implementations that do not use these common libraries
+    // first.
+    pub fn observed_external_addr(&self) -> watch::Receiver<Option<SocketAddr>> {
```

Comment:
how would this report IPv4 and IPv6 addresses, would they appear one after the other?

Reply:
Can't currently happen, there is only one path.  The most that can happen is that one peer moves (or is re-bound by a NAT) and you get a new observed address.

Reply:
just as @flub said, we would need to have multipath first and then adjust the API to surface this clearly